adage/analyze/api.py

```diff
@@ -427,7 +427,14 @@ def apply_filters(self, request, applicable_filters):
                 raise BadRequest("Invalid gene IDs: %s" % genes)
             # "in" operator in Django supports both list and set.
             qset = Q(gene1__in=ids) | Q(gene2__in=ids)
-            object_list = object_list.filter(qset).distinct()
+            direct_edges = Edge.objects.filter(qset).distinct()
+            related_genes = set()
+            for e in direct_edges:
+                related_genes.add(e.gene1)
+                related_genes.add(e.gene2)
+            object_list = object_list.filter(
+                gene1__in=related_genes, gene2__in=related_genes
+            ).distinct()
```

Comment:
True. In http://adage.greenelab.com/#/gene_network/?genes=844&mlmodel=1
when the minimum correlation is set at 0.5, 8 edges are shown between the queried gene in the URL (dnaA) and 8 other genes.

In http://165.123.67.202/#/gene_network/?genes=6546&mlmodel=4
when the minimum correlation is set at 0.5, you can see the same 8 edges that originate from dnaA, but there are other edges whose correlations are no less than 0.5 and are between two genes that are directed related to dnaA. Are you saying that the network should only show the edges that are among those 8 genes? What do you think @cgreene?